client/my-sites/site-settings/publishing-tools/hooks/use-get-post-by-voice.tsx

```diff
@@ -0,0 +1,34 @@
+import { useQuery } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+import type {
+	PostByVoice,
+	PostByVoiceResponse,
+} from 'calypso/my-sites/site-settings/publishing-tools/types';
+
+export const getPostByVoicePath = ( siteId: number | null ) => `/sites/${ siteId }/post-by-voice`;
+
+export const getCachePostByVoiceKey = ( siteId: number | null ) => [
+	'sites',
+	siteId,
+	'post-by-voice',
+];
+
+export const parsePostByVoiceResponse = ( data: PostByVoiceResponse ): PostByVoice => ( {
+	isEnabled: data.is_enabled,
+	code: data.code,
+} );
+
+export const useGetPostByVoice = ( siteId: number | null ) => {
+	return useQuery< PostByVoice >( {
+		queryKey: getCachePostByVoiceKey( siteId ),
+		queryFn: async () => {
+			const response: PostByVoiceResponse = await wpcom.req.get( {
+				path: getPostByVoicePath( siteId ),
+				apiNamespace: 'wpcom/v2',
+			} );
+
+			return parsePostByVoiceResponse( response );
```

Comment:
Here is [another option](https://github.com/Automattic/wp-calypso/blob/e1f240e1fb5a4083592a787b3e6a4f4c9f553692/client/state/data-layer/utils.js#L45) I thought I could mention:

```suggestion
			return convertToCamelCase( response );
```

Reply:
Nice idea 👍

I have created separate PR with the TS version of this function: https://github.com/Automattic/wp-calypso/pull/90988 When you get a chance, could you please review it